### DISCUSSION TITLE
About availability and support

### DISCUSSION THREAD

**Discussion by ISProjectsIoTCR** [skydiver/ewelink-api#198] [Q&A]
**Hello skydiver.**
First of all I want to congratulate you for this valuable work. I have two comments for you, and I will be very grateful if you give me an important answer.

I would like you to tell me about the availability and continuation of the ewelink API for the next years, currently it works very well but I want to develop something big and I am worried that it may soon be obsolete since there are no recent updates, what can you tell me about that?

My other query is about some devices such as switches or other sonoff models that are not retrieved by getDevices(), a few months ago you suggested using an APP_ID and an APP_SECRET but recently it doesn't work anymore. Any ideas?

**Comment by alexose** [comment]
Just added a new comment over at https://github.com/skydiver/ewelink-api/issues/88 with regards to the missing devices issue.  Might be able to make some progress there.

Ultimately though, control of the API is entirely up to the people that run Ewelink/CoolKit.  To my knowledge they have provided no guarantees that it will be accessible in the future.  (To the contrary-- They've consistently made it worse over time.)

The only futureproof solution is to flash them with your own firmware.
